### ISSUE TITLE
[BUG] Storage queue trigger overwrites message ID with id in message body

### ISSUE THREAD

**Issue by uberger** [Azure/azure-functions-python-worker#1373]
###### Please provide the following:

- Python version: 3.10.5
- Core Tools version: 4.0.5198
- Function Runtime version: 4.21.1.20667

###### Provide the steps required to reproduce the problem:

1. Create new functions app and install requirements.
2. Create new function `func new --name QueueTriggerFunc`.
3. Set the queue trigger connection in `function.json` to `AzureWebJobsStorage`.
4. Edit the function code file `__init__.py` to be
   ```python
   import logging
   import azure.functions as func

   def main(msg: func.QueueMessage) -> None:
       logging.info(f'msg.id: {msg.id}')
       logging.info('msg.body: %s',
                    msg.get_body().decode('utf-8'))
   ```
5. Start Azurite, create queue named `python-queue-items`.
6. Start Azure function host using `func start`.
7. Using Azure Storage Explorer, add a new message to the `python-queue-items` with message body:
   ```json
   {"id":"00000000-aaaa-0001-0003-00000000aaa1"}
   ```

###### Provide a description of the expected behavior.

In the console output, the `MessageId` reported by the 'Trigger Details' should match the one one reported by the line starting with `msg.id`.

###### Provide a description of the actual behavior observed.

- The `MessageId` reported by the 'Trigger Details' is the correct, generated value, e.g. `cf333465-97d3-4507-b50c-7b69308640d9`.
- The `msg.id` in the function code is overwritten with the value from the message body: `00000000-aaaa-0001-0003-00000000aaa1`

```log
[2023-11-23T10:36:48.769Z] Executing 'Functions.QueueTriggerFunc' (Reason='New queue message detected on 'python-queue-items'.', Id=faf613bc-24fb-4f69-b7bc-b75c3077beef)
[2023-11-23T10:36:48.775Z] Trigger Details: MessageId: cf333465-97d3-4507-b50c-7b69308640d9, DequeueCount: 1, InsertedOn: 2023-11-23T10:36:47.000+00:00
[2023-11-23T10:36:48.871Z] msg.body: {"id":"00000000-aaaa-0001-0003-00000000aaa1"}
[2023-11-23T10:36:48.871Z] msg.id: 00000000-aaaa-0001-0003-00000000aaa1
[2023-11-23T10:36:48.904Z] Executed 'Functions.QueueTriggerFunc' (Succeeded, Id=faf613bc-24fb-4f69-b7bc-b75c3077beef, Duration=155ms)
```

###### Provide the requirements.txt file to help us find out module related issues.

```
azure-functions
```

**Comment by bhagyshricompany** [comment]
Thanks for reporting will check  and update.

**Comment by bhagyshricompany** [comment]
pls share the req.txt  file.

**Comment by bhagyshricompany** [comment]
Hi @uberger, it's an ID that is different from the message ID because when we insert a new item, it gets generated as a new autoincrement unique ID for ref you cen see  in image.
<img width="710" alt="image" src="https://github.com/Azure/azure-functions-python-worker/assets/116645353/adf87335-d4b9-4887-9132-3e7a7eed3612">
<img width="681" alt="image" src="https://github.com/Azure/azure-functions-python-worker/assets/116645353/4ff76ce1-a8c8-4bd9-9f7e-f3e20a628321">.

**Comment by uberger** [comment]
Thanks @bhagyshricompany for looking into this. Yes, I see the same behavior and understand that the `Id` in your screenshot is auto-generated by the Azure Storage Queue service. I expect that this `Id` is also accessible through the `msg` object that gets passed into the `main` method (`msg.id`). However, it is overwritten by the `id` I defined in the message body/message text. In your example I expect in the third to last log message to see

```
[timestamp] msg.id: bbcfe0e4-e177-4b25-9f8e-1223e4b8d1d3
```

Though, it prints the `id` from the message body/message text.

Maybe worth mentioning: If I send a message that does not contain an `id` property in the JSON string it works as expected.

**Comment by bhagyshricompany** [comment]
yes you can 
<img width="661" alt="image" src="https://github.com/Azure/azure-functions-python-worker/assets/116645353/81d382e4-241f-4238-8244-86a4fbc54ee8">
<img width="704" alt="image" src="https://github.com/Azure/azure-functions-python-worker/assets/116645353/cac8e99d-f5ac-42aa-a725-9cd4d6f54a63">

**Comment by uberger** [comment]
Indeed, if I don't use `id` as a property in the JSON string (as you did in your example with `msg.id`) the Azure Storage Queue message ID is not overwritten and is accessible in the `main` function.

My problem is, that the design of the JSON string is outside of my control. The message I receive will definitely contain a JSON item with the name `id`. Then, the Azure Storage Queue message ID is lost to me. To me this doesn't appear to be intentional behavior and that's why I raised this bug report.

**Comment by hallvictoria** [comment]
Hey @uberger thanks for your patience. I looked into this issue and here are some things that I found:

- This error only occurs for Queue messages with content in dictionary / JSON format
- The message content is added as part of the object's metadata dict. When the content is in JSON format, those fields are also added as key-value pairs in the object metadata thus overwriting other attributes of the same name
- In this case, the id attribute was being overwritten, but it can also occur for any other attribute (e.g. PopReceipt)

Need the extension team to check and see how this can be addressed
